libs/driver/include/driver/GuiScale.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2005 - 2023 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "Point.h"
+#include "helpers/mathFuncs.h"
+#include <type_traits>
+
+/// Represents the scale factors – precomputed from a percentage – to translate between screen and view dimensions
+class GuiScale
+{
+public:
+    constexpr GuiScale(unsigned percent) noexcept
+        : percent_(percent), scaleViewToScreen_(percent_ * .01f), scaleScreenToView_(1.f / scaleViewToScreen_)
```

Comment:
No need for manual optimizations, be explicit:
```suggestion
        : percent_(percent), scaleViewToScreen_(percent_ / 100f), scaleScreenToView_(1.f / scaleViewToScreen_)
```

Reply:
I know `-ffast-math` does it for me, but it's a hard habit to break.

Reply:
Even if it doesn't the performance penalty for our occasional use of floats is negligible and readability is more important. But yes, I know that force of habbit ;-)